### DISCUSSION TITLE
Access ddev project database from separate docker container?

### DISCUSSION THREAD

**Discussion by juppwerner** [ddev/ddev#5418] [Q&A]
Hello, 

I have running a ddev project including a MySQL database successfully.

I want to get the schema documentation of the database using another docker image from my host like this:

$ docker run \\
  --mount type=bind,source="$(pwd)",target=/home/schcrwlr/share \\
  --rm -it \\
  schemacrawler/schemacrawler \\
  /opt/schemacrawler/bin/schemacrawler.sh \\
  --server=mysql \\
  --host=localhost \\
  --port=59002 \\
  --database=db \
  --user=db \\
  --password=db \\
  --info-level=standard \\
  --command script \\
  --title "MyDatabase Schema" \\
  --script-language python \\
  --script plantuml.py

(similar to: https://dev.to/sualeh/generate-good-looking-plantuml-diagrams-for-your-database-3g19)

However the output says:

    Error: Could not connect to <jdbc:mysql://localhost:59002/db>, for <user 'db'>, with properties <{}>

I used the DB port as printed using `ddev describe`. I also tried host=127.0.0.1, and ddev-myproject-db, same result.
I also tried user/password: root/root.

Any hints appreciated!

Thank you.

regards,
Joachim

**Comment by rfay** [comment]
If you're running another docker container, you'll have to have it on the `ddev_default` network, and it will access the ddev database container using `mysql://db:db@ddev-<projectname>-db/db` 

For communicating inside the docker network: https://ddev.readthedocs.io/en/latest/users/usage/faq/#can-different-projects-communicate-with-each-other

You could also install the schema crawler in the web or db containers, etc.

**Reply by juppwerner** [reply]
Hello rfay,
thank you. 
I tried adding " ... --network=ddev_default", still no connection.

What would have to be used as --host and --port here?

Just to clarify, the above schemacrawler docker run ist not belonging to a ddev project on my host, just an adhoc docker run ...

Thank you.

**Reply by rfay** [reply]
This seems to work for the running project named `d10`: 
`$ docker run --mount type=bind,source="$(pwd)",target=/Users/rfay/tmp/output --rm -it --network=ddev_default schemacrawler/schemacrawler /opt/schemacrawler/bin/schemacrawler.sh --server=mysql --host=ddev-d10-db --port=3306 --database=db --user=db --password=db --info-level=standard --title="MyDatabaseSchema" --script-language=python  --command=script --script plantuml.py`

Note that the image in question is built only for amd64 machines (Intel). So if you have big plans for this on macOS they may not go well.
